tests/src/test/scala/com/nvidia/spark/rapids/QueryFuzzer.scala

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids
+
+import java.util.concurrent.atomic.AtomicLong
+
+import scala.util.Random
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.{DataTypes, DecimalType}
+
+class QueryFuzzer(val seed: Long) {
+
+  val rand = new Random(seed)
+
+  val idGen = new AtomicLong(0)
```

Comment:
Curious on the choice to use an `AtomicLong` here.  If we have threads racing through this code, how can we get a deterministic plan for a given seed?  If we don't have racing threads, why do we need `AtomicLong`?

Reply:
This was really just for convenience. I can implement a generator method in the class instead.